main/src/cgeo/geocaching/settings/SettingsActivity.java

```diff
@@ -309,8 +309,18 @@ private void initServicePreferences() {
 
     private void initLCServicePreference(final boolean gcConnectorActive) {
         final boolean isActiveGCPM = gcConnectorActive && Settings.isGCPremiumMember();
-        getPreference(R.string.preference_screen_lc).setSummary(getServiceSummary(Settings.isLCConnectorActive() && isActiveGCPM));
-        getPreference(R.string.pref_connectorLCActive).setEnabled(isActiveGCPM);
+        getPreference(R.string.preference_screen_lc).setSummary(getLcServiceSummary(Settings.isLCConnectorActive(), gcConnectorActive));
+        if (isActiveGCPM) {
+            getPreference(R.string.pref_connectorLCActive).setEnabled(true);
+        }
+    }
+
+    private String getLcServiceSummary(final boolean lcConnectorActive, final boolean gcConnectorActive) {
+        return !lcConnectorActive ? StringUtils.EMPTY :
```

Comment:
I find this statement overwhelmingly complex but this might just be the limitation of my brain built-in Java parser.

Reply:
Are you referring explicitly to the method returning the Lc connector state summary?

Basically it ads to the former two-way-state (active or not active) a third state "active/disabled" (in case LC connector is set to active but can't be used because the logged in user is not premium)

Reply:
> Are you referring explicitly to the method returning the Lc connector state summary?
> 
> Basically it ads to the former two-way-state (active or not active) a third state "active/disabled" (in case LC connector is set to active but can't be used because the logged in user is not premium)

No, it is the multiline return statement in line 319 that I have a hard time to parse. It is the nested use of the ternary operator. In order to truely understand the logic behind it, I'd have to unravel it into if / then / else statements and using simple expressions / variables ... which I haven't done in the context of this review, instead, I just assume that it works the way you intend ;)

Reply:
@baiti That's a fair assessmemt and it clearly shows that my code was too complex. I cleaned it up, may you take a second look at the new version